e2e/cypress/integration/files_and_attachments/upload_files_spec.js

```diff
@@ -202,6 +205,83 @@ describe('Upload Files', () => {
+        cy.postMessage('Post with attachment to be deleted');
+
+        // # Get the last post
+        cy.getLastPostId().then((lastPostId) => {
+            // # Scan inside of the last post message
+            cy.get(`#${lastPostId}_message`).and('be.visible').within(() => {
+                // * Check if the attached image is in the post and then click to open the preview
+                cy.findByLabelText(`file thumbnail ${attachmentFilename}`).should('be.visible').click();
+            });
+        });
+
+        // * Verify preview modal is opened
+        cy.get('.a11y__modal').should('exist').and('be.visible').
+            within(() => {
+            // * Check if get public link button is present and click it
+                cy.findByText('Get a public link').should('be.visible').click({force: true});
+            });
+
+        // # Wait a little for link to get generate
+        cy.wait(TIMEOUTS.ONE_SEC);
+
+        // * Verify copy public link modal is opened
+        cy.get('.a11y__modal.modal-dialog').should('exist').and('be.visible').
+            within(() => {
+                // * Verify that copy link button is present
+                cy.findByText('Copy Link').should('be.visible');
+
+                // # Get the copy link of the attachment and save for later purpose
+                cy.get('#linkModalTextArea').invoke('text').as('publicLinkOfAttachment');
+            });
+
+        // # Close the image preview modal
+        cy.get('body').type('{esc}');
+
+        // # Once again get the last post with attachment, this time to delete it
+        cy.getLastPostId().then((lastPostId) => {
+            // # Click post dot menu in center.
+            cy.clickPostDotMenu(lastPostId);
+
+            // # Scan inside the post menu dropdown
+            cy.get(`#CENTER_dropdown_${lastPostId}`).should('exist').within(() => {
+                // # Click on the delete post button from the dropdown
+                cy.findByText('Delete').should('exist').click();
+            });
+        });
+
+        // * Verify caution dialog for delete post is visible
+        cy.get('.a11y__modal.modal-dialog').should('exist').and('be.visible').
+            within(() => {
+                // # Confirm click on the delete button for the post
+                cy.findByText('Delete').should('be.visible').click();
+            });
+
+        // # Try to fetch the url of the attachment we previously deleted
+        cy.get('@publicLinkOfAttachment').then((publicLinkOfAttachment) => {
+            cy.request({url: publicLinkOfAttachment, failOnStatusCode: false}).then((response) => {
+                // * Verify that the link no longer exists in the system
+                expect(response.status).to.be.equal(404);
+
+                // * Verify that it contains text stating unable to get the file info
+                expect(response.body).to.contain('Unable+to+get+the+file+info.');
+            });
```

Comment:
Thanks for updating. Could you add action/check below and see if it works for you?
```javascript
// # Visit the public link
cy.visit(publicLinkOfAttachment, {failOnStatusCode: false});

// * Page should display an error
cy.findByText('Error').should('be.visible');
cy.findByText('Unable to get the file info.').should('be.visible');
cy.findByText('Back to Mattermost').should('be.visible').parent().should('have.attr', 'href', '/');
```